src/App.jsx

```diff
@@ -17,22 +18,33 @@ const tokenKey = 'token';
 
 const App = () => {
   const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const { isLoggedIn, role } = useSelector((state) => state.login);
 
   useEffect(() => {
     if (localStorage.getItem(tokenKey)) {
-      dispatch(login(JSON.parse(localStorage.getItem(tokenKey))));
+      dispatch(loginActions.login(JSON.parse(localStorage.getItem(tokenKey))));
     }
 
-    dispatch(getAllRoomsAsync());
+    dispatch(getAllCities());
```

Comment:
- [ ] I don't get the idea of fetching cities when you start the application, I understand that for your case is necessary to add a new hotel, taking that into account, it would be better to have the `useEffect` hook when mounting the `AddHotel` component and there dispatch this action. Take into account that right now this action is going to be dispatched every time you render a different route.